### PR TITLE
Developer guide use cases

### LINKED ISSUE #70: Add the delete and edit for skill or achievement, and switch tab use cases in the Developer Guide for v1.1

### PR DESCRIPTION
**Pull request by seanjyjy**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -86,6 +86,55 @@ Guarantees: Editing of internship application is successful
...
+Precondition: User already has an existing list of skills and achievements. </br>
+Guarantees: Deletion of skill or achievement is successful
+
+1.  User requests to delete a skill or achievement from the user profile and provides the index.
+2.  InternHunter removes the skill or achievement from the user profile. </br>
+    Use case ends.
+
+#### Extensions
+
+1a. InternHunter detects that the index entered is out of bounds. </br>
```
Comment (orzymandias) [inline docs/DeveloperGuide.md]: Do you think it is good to be overly specific about the type of error? If so then there are many more extensions to include such as inputting an index that is not a number for example.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -86,6 +86,55 @@ Guarantees: Editing of internship application is successful
...
+
+#### Extensions
+1a. InternHunter detects that the index entered is out of bounds. </br>
+  1a1. InternHunter displays an error message and informs the user of the valid input format. </br>
+  Use case resumes from step 1.
+  
+1b. InternHunter detects an error in the input format. </br>
+  1b1. InternHunter displays an error message and informs the user of the valid input format. </br>
+  Use case resumes from step 1.
+
```
Comment (orzymandias) [inline docs/DeveloperGuide.md:121]: I am unsure about this myself but what if there are no items in the list? what kind of interactions can the user and system have? I know your preconditions addresses that but do we have to consider the case where you call edit and there are no items in the list?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (seanjyjy) [reply docs/DeveloperGuide.md]: Yup we could keep it simple currently and once we merge them in the future, we could include those extensions.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (seanjyjy) [reply docs/DeveloperGuide.md:121]: perhaps once we merge the similar use cases together, we can include more extensions to it!

**Review by orzymandias — APPROVED** [review]
LGTM

**Pull request merged**